runtime/bcverify/vrfyhelp.c

```diff
@@ -904,6 +904,16 @@ isFieldAccessCompatible(J9BytecodeVerificationData *verifyData, J9ROMFieldRef *f
 
 	if (JBputfield == bytecode) {
 		if (J9_ARE_ALL_BITS_SET(receiver, BCV_SPECIAL_INIT)) {
+#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
+			J9BranchTargetStack *liveStack = (J9BranchTargetStack *)verifyData->liveStack;
+			if (FALSE == liveStack->uninitializedThis) {
+				/* ACC_STRICT field must be assigned before instance initialization method. */
+				J9ROMFieldShape *field = findFieldFromCurrentRomClass(romClass, fieldRef);
```

Comment:
You'd better consolidate these assignments to the front of this function to avoid duplicating them everywhere.

e.g.
```
isFieldAccessCompatible(J9BytecodeVerificationData *verifyData, J9ROMFieldRef *fieldRef, UDATA bytecode, UDATA receiver, IDATA *reasonCode)
{
	J9ROMClass *romClass = verifyData->romClass;
	J9ROMConstantPoolItem *constantPool = (J9ROMConstantPoolItem *)(romClass + 1);
	J9UTF8 *utf8string = J9ROMCLASSREF_NAME((J9ROMClassRef *)&constantPool[fieldRef->classRefCPIndex]);
J9BranchTargetStack *liveStack = (J9BranchTargetStack *)verifyData->liveStack;
J9ROMFieldShape *field = findFieldFromCurrentRomClass(romClass, fieldRef);
...

```

Reply:
Thanks Cheng, I added your suggestions.